src/sparseml/pytorch/datasets/classification/_ffcv/__ffcv.py

```diff
@@ -0,0 +1,234 @@
+        )
+        self.transform = transform
+
+        with self.switched_off_transform():
+            self._write(num_workers=num_workers)
+
+        self._image_size = image_size
+        self._device = device
+
+    def _write(self, num_workers: int = 16):
+        """
+        Writes the dataset to the write_path.
+        """
+        assert ffcv, "FFCV is not installed"
+        writer = DatasetWriter(
+            self.write_path,
+            {
+                "image": RGBImageField(),
+                "label": IntField(),
+            },
+            num_workers=num_workers,
+        )
+
+        LOGGER.info(f"Writing dataset to {self.write_path}")
+        writer.from_indexed_dataset(self.dataset)
+
+    @contextmanager
+    def switched_off_transform(self):
+        """
+        Context manager to switch off the transforms in original dataset,
+        and then switch it back on after the context is exited.
+        """
+        _old_transform = None
+        _old_target_transform = None
+
+        try:
+            # Switch off the transforms
+            if hasattr(self.dataset, "transform"):
+                _old_transform = self.dataset.transform
+                self.dataset.transform = None
+                LOGGER.info("Transforms switched off for original dataset.")
+
+            if hasattr(self.dataset, "target_transform"):
+                _old_transform = self.dataset.target_transform
+                self.dataset.transform = None
+                LOGGER.info("Target Transforms switched off for original dataset.")
+
+            # Give back control
+            yield
+
+        finally:
+            # restore the transforms
+            if _old_transform is not None and hasattr(self.dataset, "transform"):
+                self.dataset.transform = _old_transform
+                LOGGER.info("Transforms switched on for original dataset.")
+
+            if _old_target_transform is not None and hasattr(
+                self.dataset, "target_transform"
+            ):
+                self.dataset.target_transform = _old_target_transform
+                LOGGER.info("Target Transforms switched on for original dataset.")
+
+    def get_loader(
+        self,
+        num_workers: int = 16,
+        batch_size: int = 32,
+        distributed: int = 0,
+        in_memory: int = 0,
+        validation: bool = False,
+    ):
+        """
+        Initialize the data loader.
+
+        :param num_workers: Number of workers to use for the data loader.
+        :param batch_size: Batch size for the data loader.
+        :param distributed: 1/0 Whether to use distributed data loading.
+        :param in_memory: 1/0 Does the dataset fit in memory.
+        """
+
+        _init_msg = "Initializing data loader."
+        if validation:
+            _init_msg = "Initializing validation data loader."
+        LOGGER.info(_init_msg)
+
+        resolution = (self._image_size, self._image_size)
+        decoder = (
```

Comment:
need to extensively check that for our image classification pipelines these are equivalent to our current transforms